src/core/MOM_stoch_eos.F90

```diff
@@ -0,0 +1,212 @@
+!> Provides the ocean stochastic equation of state
+module MOM_stoch_eos
+! This file is part of MOM6. See LICENSE.md for the license.
+use MOM_grid,            only : ocean_grid_type
+use MOM_hor_index,       only : hor_index_type
+use MOM_file_parser,     only : get_param, param_file_type
+use MOM_random,          only : PRNG,random_2d_constructor,random_2d_norm
+use MOM_time_manager,    only : time_type
+use MOM_io,              only : vardesc, var_desc
+use MOM_restart,         only : MOM_restart_CS,is_new_run
+use MOM_diag_mediator,   only : register_diag_field,post_data,diag_ctrl,safe_alloc_ptr
+use MOM_variables,       only : thermo_var_ptrs
+use MOM_verticalGrid,    only : verticalGrid_type
+use MOM_restart,         only : register_restart_field
+use MOM_isopycnal_slopes,only : vert_fill_TS
+!use random_numbers_mod, only : getRandomNumbers,initializeRandomNumberStream,randomNumberStream
+
+implicit none
```

Comment:
The MOM6 style guide section on implicit variables and public types (https://github.com/mom-ocean/MOM6/wiki/Code-style-guide#implicit-variables) requires that all modules should declare `implicit none ; private`.  This would then allow any remaining private statements to be removed, and require an explicit public declaration on any public types.